packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -983,35 +905,25 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
      * @param tokenId   Token Id of the KGH.
      * @param kroInKgh  The amount of KRO in the KGH.
      * @param kroShares The amount of KRO shares to receive.
-     * @param kghShares The amount of KGH shares to receive.
      */
     function _delegateKgh(
         address validator,
         uint256 tokenId,
         uint128 kroInKgh,
-        uint128 kroShares,
-        uint128 kghShares
+        uint128 kroShares
     ) internal {
-        Vault storage vault = _vaults[validator];
-        Asset storage asset = vault.asset;
-
-        KGH.safeTransferFrom(msg.sender, address(this), tokenId);
+        Asset storage asset = _vaults[validator].asset;
+        KghDelegator storage kghDelegator = _vaults[validator].kghDelegators[msg.sender];
```

Comment:
How about getting `address delegator` as an input instead of `msg.sender`? Also in `_delegateKghBatch`

Reply:
It's better. I'll make change as you suggested.